hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/entity/AbstractNode.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.common.domain.entity;
+
+import com.google.common.collect.Range;
+import com.hedera.mirror.common.domain.History;
+import com.hedera.mirror.common.domain.Upsertable;
+import jakarta.persistence.Column;
+import jakarta.persistence.MappedSuperclass;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+@Data
+@MappedSuperclass
+@NoArgsConstructor
+@SuperBuilder(toBuilder = true)
+@Upsertable(history = true, skipPartialUpdate = true)
+public abstract class AbstractNode implements History {
+
+    @ToString.Exclude
+    private byte[] adminKey;
+
+    @Column(updatable = false)
+    private long createdTimestamp;
+
+    private boolean deleted;
+
+    @jakarta.persistence.Id
```

Comment:
nit, since there is no longer name clash with the deleted `Id` class

```suggestion
    @Id
```